src/track/beatsimporter.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <QVector>
+#include <memory>
+
+#include "audio/streaminfo.h"
+
+namespace mixxx {
+
+/// Importer class for Beats that can correct timing offsets when the
+/// signal info (channel number, sample rate, bitrate) is known.
+class BeatsImporter {
+  public:
+    BeatsImporter() = default;
+    virtual ~BeatsImporter() = default;
+
+    virtual bool isEmpty() const {
+        return true;
+    };
+
+    /// Determines the timing offset and returns a Vector of frame positions
+    /// to use as input for the BeatMap constructor
+    virtual QVector<double> importBeatsWithCorrectTiming(
```

Comment:
> Which timing is correct depends on the use case.

What do you mean? The timing of *imported* beats/cues is correct if and only if the beats are at the exact same positions as in Serato. Or what do you mean?

> The function also dos not the import, it reads and returns a adjusted beats vector.
>
> So how about getBeatsWithMixxxTiming() or such.

I think the naming for beats and cue import should be consistent, and the "import" name was requested by @uklotzde here: https://github.com/mixxxdj/mixxx/pull/2707#discussion_r415381229

Reply:
I think I disagree here with @uklotzde, because import implies also storing. This function does not store anything. 
But there is no strong need to change it. 

> The timing of imported beats/cues is correct if and only if the beats are at the exact same positions as in Serato

Or 

> The timing of imported beats/cues is correct if they match the values in Serato. 

We have no definition of "CorrectTiming"

Reply:
> The timing of imported beats/cues is correct if they match the values in Serato.

They already match the values in Serato without this function. This function corrects the offset that is caused by Mixxx using a different decoder. Unfortunately there is no "Mixxx" timing as is depends on the decoder that is currently used.

Reply:
OK, so it might become:

importBeatsWithOffsetForCurrentEncoder() 
 
?

Reply:
I renamed it to "importBeatsAndApplyTimingOffset".